.github/workflows/icu4c.yml

```diff
@@ -240,18 +266,237 @@ jobs:
           CPPFLAGS: -fsanitize=thread
           LDFLAGS: -fsanitize=thread
 
+  #------------------------------------------------------
+  # Clang Linux with data filter
+  # Note: This job uses `make -j2` instead of `make -j -l4.5` because with more parallelism (PR #2456)
+  # this check became flaky. The build apparently was not done copying one or another .ucm file before
+  # calling makeconv for it, although the Makefile has appropriate dependencies.
+  clang-datafilter:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      - name: Build ICU4C with clang
+        run: |
+          cd icu4c/source && \
+          ICU_DATA_FILTER_FILE=../../.github/data-filter.json ./runConfigureICU Linux/clang && \
+          make -j2 tests && \
+          \[ ! -d data/out/build/icudt66l/translit \] && \
```

Comment:
Why are we referring to "icudt66l", the data folder from ICU 66?

Reply:
Updated to use latest icu major version